scripts/_common.sh

```diff
@@ -199,15 +200,58 @@ function generate_matrix() {
     for client_version in "${versions[@]}"; do
       for gateway_version in "${versions[@]}"; do
         if "$filter_fn" "$fed_version" "$client_version" "$gateway_version"; then
+
           # bash doesn't allow returning arrays, however we can mimic the
           # behavior of returning an array by echoing each element
-          echo "FM: $fed_version CLI: $client_version GW: $gateway_version"
+          if are_all_versions_current $fed_version $client_version $gateway_version; then
+              echo "FM: $fed_version CLI: $client_version GW: $gateway_version LNv2: 1"
+          elif supports_lnv2 $fed_version $client_version $gateway_version; then
+            for enable_lnv2 in {0..1}; do
+              echo "FM: $fed_version CLI: $client_version GW: $gateway_version LNv2: $enable_lnv2"
+            done
+          else
+              echo "FM: $fed_version CLI: $client_version GW: $gateway_version LNv2: 0"
+          fi
         fi
       done
     done
   done
 }
 
+export LNV2_STABLE_VERSION="v0.6-rc.1"
+function version_lt() {
```

Comment:
I think `sort -V` will incorrectly sort for `rc`s

```
echo -e "v0.4.4\nv0.4.4-rc.2" | sort -V
v0.4.4
v0.4.4-rc.2
```

but we can `s/-/~` and that should fix this sorting

```
echo -e "v0.4.4\nv0.4.4-rc.2" | sed 's/-/~/g' | sort -V
v0.4.4~rc.2
v0.4.4
```

We can also simplify the `current` checks. Perhaps we can rewrite `version_lt` something like this

```bash
function version_lt() {
  if [ "$1" = "current" ]; then
    return 1
  elif [ "$2" = "current" ]; then
    return 0
  fi

  # replace `-` with `~` so `sort -V` correctly sorts pre-releases
  v1="${1//-/\~}"
  v2="${2//-/\~}"

  [ "$v1" != "$(echo -e "$v1\n$v2" | sort -V | tail -n 1)" ]
}
```